enhancements/network/user-defined-network-for-virt-workloads.md

```diff
@@ -0,0 +1,563 @@
+
+  Examples (metrics, alerts, operator conditions)
+  - authentication-operator condition `APIServerDegraded=False`
+  - authentication-operator condition `APIServerAvailable=True`
+  - openshift-authentication/oauth-apiserver deployment and pods health
+
+- What impact do these API extensions have on existing SLIs (e.g. scalability, API throughput,
+  API availability)
+
+  Examples:
+  - Adds 1s to every pod update in the system, slowing down pod scheduling by 5s on average.
+  - Fails creation of ConfigMap in the system when the webhook is not available.
+  - Adds a dependency on the SDN service network for all resources, risking API availability in case
+    of SDN issues.
+  - Expected use-cases require less than 1000 instances of the CRD, not impacting
+    general API throughput.
+
+- How is the impact on existing SLIs to be measured and when (e.g. every release by QE, or
+  automatically in CI) and by whom (e.g. perf team; name the responsible person and let them review
+  this enhancement)
+
+- Describe the possible failure modes of the API extensions.
+- Describe how a failure or behaviour of the extension will impact the overall cluster health
+  (e.g. which kube-controller-manager functionality will stop working), especially regarding
+  stability, availability, performance and security.
+- Describe which OCP teams are likely to be called upon in case of escalation with one of the failure modes
+  and add them as reviewers to this enhancement.
+
+## Support Procedures
+
+N/A
+
+## Alternatives
+
+### Binding mechanism
+
+We could adapt the existing (in-tree) bridge binding to bind to the UDN
+interface, and use it instead of passt. This would mimic the approach taken
+both for secondary layer2 and localnet networks (which use
+[persistent IPs](https://github.com/openshift/enhancements/pull/1456)) and
+also HyperShift (which relies on
+[point to point routing](https://github.com/openshift/enhancements/blob/master/enhancements/network/ovn-hypershift-live-migration.md#topology)).
+Doing so would solve the TCP connection reset issue we currently face on passt.
+
+If we implemented
+[this feature request](https://issues.redhat.com/browse/CNV-32573) in
+OVN-Kubernetes we have an extremely simple in-tree binding that would be
+responsible for layer2 connectivity: OVN-Kubernetes would provision DHCP flows
+for the VMs in the logical switches implementing the flat layer2 networks. To
+integrate with network policies pod / namespace selectors we would need to have
+a way for the pod to report the IPs effectively "owned" by the VM.
+
+There are some drawbacks to bridge binding though, namely:
+- would currently only work for IPv4. We would need to advertise routes either
+from OVN-Kubernetes, or from within the pod (which would require CAP_NET_ADMIN,
+which we do **not** want to reintroduce).
+- integrating with pod network ecosystem (service mesh, metrics, etc) difficult
```

Comment:
Do you remember if mac spoofing mechanism was not working with linux bridge or was the opposite ? Don't know if it would work for passt though.

Reply:
I don't know what you mean.

Reply:
There is a mac spoofing mechanism implemented at kubevirt bindings, don't know if we miss that with passt or not.

Reply:
no, there's MAC spoofing implemented at bridge CNI. 

There's also MAC (and IP) spoofing implemented in OVN-K so we do more.